File: Programmers/level2/findPrime/find_prime.py
# 문제 출처 : https://programmers.co.kr/learn/courses/30/lessons/42839?language=python3

from itertools import permutations
import math


def solution(numbers):
    answer = 0
    '''
    오류코드
    arr = list(map(int, list(numbers)))
    for per in permutations(list(numbers)):
        arr.append(int(''.join(per)))
    print(set(arr))
    배열에 제대로 순열이 안들어감
    '''
    new_numbers = list(numbers)
    for i in range(2, len(numbers)+1):
        per = list(permutations(numbers, i))
        for j in per:
            if len(j) <= len(numbers):
                new_numbers.append(''.join(j))
    # print(new_numbers)
    new_numbers = list(set([int(x) for x in new_numbers]))
    # print(new_numbers)
    for num in new_numbers:
        if primetester(num) == True:
            answer += 1
    return answer


def primetester(n):
    if n <= 0 or n % 1 != 0:
        return False
    elif n == 1:
        return False
    else:
        num = math.sqrt(n)
        for i in range(2, int(num + 1)):
            if n % i == 0:
                return False
    return True

File: Programmers/level2/findPrime/findPrime.js
// 문제 출처 : https://programmers.co.kr/learn/courses/30/lessons/42839?language=javascript

export const solution = (numbers) => {
    var answer = 0;
    let n = numbers.split('')
    let nums = new Set();
    permutation(n,'');
    
    function permutation(a,s){
        if(s.length > 0){
            if(nums.has(Number(s)) === false){
                nums.add(Number(s));
                if(primeTester(Number(s))){
                    answer++;
                }
            }
        }
        if(a.length > 0){
            for(let i = 0; i < a.length; i++){
                let t = a.slice(0);
                console.log("t : ", t)
                t.splice(i, 1);
                permutation(t, s + a[i])
            }
        }
    }
    function primeTester(n){
        if (typeof n !== "number" || n < 1 || n % 1 !== 0) {
    return false;
  }
  if (n === 1) {
    return false;
  }
  let num = Math.sqrt(n);
  for (let i = 2; i <= num; i++) {
    if (n % i === 0) {
      return false;
    }
  }
  return true;
    }
    console.log(nums)
    return answer;
}